enhancements/installer/operators-check.md

```diff
@@ -0,0 +1,174 @@
+issues to the appropriate operator development team.
+
+### User Stories
+
+As an openshift-install user, I want the Installer to tell me when an operator never stops
+progressing so that I can check whether the operator has an issue.
+
+As a member of the technical release team, I want the Installer to exit non-zero when
+an operator never stops progressing so that I can triage operator bugs.
+
+### Goals
+
+* Installer correctly identifies a failed cluster install where cluster operators are not stable
+* Installer exits non-zero in those situations and delivers meaningful error message
+
+### Non-Goals
+
+* Installer handling other classes of errors, such as failure to provision machines by MAO
+
+## Proposal
+
+### Workflow Description
+
+Cluster admin will begin installing cluster as usual. The Installation workflow will be:
+
+1. Cluster initializes as usual
+2. As usual, installer checks that cluster version is Available=true Progressing=False Degraded=False 
+3. Installer checks status of each cluster operator for stability
+4. If a cluster operator is not stable the installer logs a message and throws an error
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+The important detail is how we define _stable_. The definition proposed here is:
+
+```
+If a cluster operator does not maintain Progressing=False for at least 30 seconds,
+during a five minute period it is unstable.
+```
+
+### Risks and Mitigations
+
+One risk would be a false positive: the Installer identifies that a cluster
+operator is unstable, but it turns out the operator is perfectly healthy;
+the install was declared a failure but was actually successful. This risk
+seems low and a risk that could be managed by monitoring these specific failures
+in CI.
+
+### Drawbacks
+
+This enhancement has some significant potential drawbacks in terms of design:
+
+A fundamental design principle of 4.x and the operator framework has been
+delegating responsibility away from the Installer to allow clean separation
+of concerns and better maintainbility. Without this enhancement, it is the
+responsibility of the Cluster-Version Operator to determine whether given
+Cluster Operators constitute a successful version. The idea of keeping the
+cluster-version operator as the single responsible party is discussed in the
+alternatives section.
+
+Following the previous point, this enhancement could introduce a slippery slope.
+Where before this enhancement, the Installer would simply query the CVO; now we
+are adding additional conditional logic. Introducing this conditional logic leads
+to the risk of further conditions, which could make the install process more brittle
+and introduce the potential for false positives or other failures.
+
+Does implementing this enhancement address symptoms of issues with operator status definitions?
+Should operators themselves be setting Degraded=True when they don't meet this stability criterion?
```

Comment:
I don't think so.  Progressing is about moving from one steady state to another.  Degraded is about "something is broken".  They serve different purposes and some progressing states easily last longer than 5 minutes.  For instance, rolling out a new kube-apiserver level takes 15-20 minutes.